kbase-extension/static/kbase/js/userMenu.js

```diff
@@ -57,61 +57,62 @@ define([
-                                target: '_blank',
-                                'data-menu-item': 'userlabel',
-                            },
-                            [
-                                div(
-                                    {
-                                        style:
-                                            'display:inline-block; width: 34px; vertical-align: top;',
-                                    },
-                                    [
-                                        span({
-                                            class: 'fa fa-user',
-                                            style: 'font-size: 150%; margin-right: 10px;',
-                                        }),
-                                    ]
-                                ),
-                                div(
-                                    {
-                                        style: 'display: inline-block',
-                                        'data-element': 'user-label',
-                                    },
-                                    [displayName, br(), i({}, userName)]
-                                ),
-                            ]
-                        ),
-                    ]),
-                    li({ class: 'divider' }),
-                    li({}, [
-                        a({ href: '#', 'data-menu-item': 'logout', id: 'signout-button' }, [
-                            div({ style: 'display: inline-block; width: 34px;' }, [
-                                span({
-                                    class: 'fa fa-sign-out',
-                                    style: 'font-size: 150%; margin-right: 10px;',
-                                }),
+            const menu = div(
+                { class: 'dropdown', style: 'display:inline-block', dataTestid: 'userMenu' },
+                [
+                    button(
+                        {
+                            type: 'button',
+                            class: 'btn btn-default dropdown-toggle',
+                            'data-toggle': 'dropdown',
+                            'aria-expanded': 'false',
+                        },
+                        [renderAvatar(), span({ class: 'caret', style: 'margin-left: 5px;' })]
+                    ),
+                    ul({ class: 'dropdown-menu', role: 'menu' }, [
+                        li({}, [
+                            a(
+                                {
+                                    href: '/#people/' + userName,
+                                    target: '_blank',
+                                    'data-menu-item': 'userlabel',
+                                },
+                                [
+                                    div(
+                                        {
+                                            style:
+                                                'display:inline-block; width: 34px; vertical-align: top;',
```

Comment:
can all these inline styles be moved into the relevant stylesheet?

Reply:
But the goal was not to make as few code changes as possible, whilst adding testing attributes.

Reply:
Yes, they could be. But the point of changing that file at all was to make a small change to allow it to be testable, not to refactor it.